src/contracts/atlas/Atlas.sol

```diff
@@ -220,20 +213,87 @@ contract Atlas is Test, Factory {
             }
         }
 
-        // Only verify signatures of meta txs if the original signer isn't the bundler
-        // TODO: Consider extra reentrancy defense here?
-        if (dAppOp.approval.from != msg.sender && !_verifyDApp(dConfig, dAppOp)) {
-            bool bypass = isSimulation && dAppOp.signature.length == 0;
-            if (!bypass) {
+        // bundler checks
+        // user is bundling their own operation - always allowed, check valid dapp signature/callchainhash
```

Comment:
I don't think it is safe to assume that the user is always allowed to bundle their own operation

Reply:
Should that be put as a flag for the protocol to set?